src/library/scala/Unit.scala

```diff
@@ -31,6 +31,9 @@ final abstract class Unit private extends AnyVal {
 object Unit extends AnyValCompanion {
 
   /** Transform a value type into a boxed reference type.
+   *
+   *  The result of `(x : Any)` where `x` is of type `Unit` is `scala.runtime.BoxedUnit.UNIT`.
+   *  This method is not intended for use in source code, which can use `UNIT` directly.
```

Comment:
Isn't scala.runtime unstable?

Reply:
Most of us are a little unstable. The comment on the ticket was to move the API to runtime. It's necessarily an implementation detail.

Reply:
Because it's unit or because it's boxed?  It's nice to have a stable API for the other primitive value types.

Reply:
The doc for `Unit` says there is no object representation, but the caveat is the boxed representation. So to maintain that fiction, boxed unit must be more forbidden, which is OK because it's not useful. Making boxed ints is arguably useful.

Reply:
I wish these method stayed and the ()/Unit issue were more directly addressed than making these methods unreachable (or, with reflection, less reachable)

Reply:
Do you have a method somewhere with a type param and you're just itching to pass it the unit value, but you need to box it first from Java?

Reply:
Source code should use `(x: Any)`, *never* directly reference `scala.runtime`. The latter is an implementation detail of the language on some platforms. For example JS doesn't even really have `BoxedUnit` at all.

Reply:
The other types refer to runtime and suggest looking at the source code. I tuned down the language here, though I think it doesn't help anyone with interop questions.

Reply:
What are the element values of `Seq(1, 2, 3) map println)` in Scala.js?

The fact that methods returning unit don't actually require a unit value (let's say, at least on the JVM) feels kind of besides the point.  Unless you're specialising a type param to optimise for Unit you need an actual value to use.

Using `(x: Any)`, particularly with the unenforceable "where `x` is of type `Unit`" condition, seems like a less straightforward, roundabout way to achieve the same goal.  I'd rather a straightforward way, and deal with `()` vs `Unit` head on.

Reply:
OK, I've updated it to say the least helpful thing that addresses the issue, which is to document that you're not supposed to call these methods.

Reply:
> What are the element values of `Seq(1, 2, 3) map println` in Scala.js?

They are `()`. Scala.js does not box its primitive values. Well OK, they're the primitive JS value `undefined`, but that's equivalent because in Scala.js `()` and `undefined` are the same thing.